### PR TITLE
[AMORO-2316] The Files page supports filtering by partition name and sorting by dictionary value

### LINKED ISSUE #2316: [Improvement]: The Files page supports filtering by partition name and sorting by dictionary value.
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/amoro/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

When I want to check the file status of today's partition, I find that the partitions on the Files page are distributed in a disorganized way, which makes it impossible for me to query the file status of the partition I want to check.
image

### How should we improve?

Sort the partitions in dictionary order and add a search box to the page

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by shendanfengg**
## Why are the changes needed?

This is the backend implementation of #2316 .

## Brief change log

- Sort by partition name and add keywords filter

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Review by HuangFru — COMMENTED** [review]
Seems that lost frontend code?

**Comment by shendanfengg** [comment]
> Seems that lost frontend code?

The front-end code needs to be provided by another pr

File: ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/TableController.java [hunk 1]
```diff
@@ -423,12 +424,18 @@ public void getTablePartitions(Context ctx) {
     String catalog = ctx.pathParam("catalog");
     String database = ctx.pathParam("db");
     String table = ctx.pathParam("table");
+    String keywords = ctx.queryParamAsClass("keywords", String.class).getOrDefault("");
```
Comment (wangtaohz) [inline ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/TableController.java:427]: This Http API is `/tables/catalogs/{catalog}/dbs/{db}/tables/{table}/partitions?keywords=xxxxx`.

How about changing `keywords` to `filter`?

**Review by wangtaohz — APPROVED** [review]
LGTM

**Review by zhoujinsong — APPROVED** [review]
LGTM.
Thanks a lot for you contribution.

**Comment by zhoujinsong** [comment]
Issue #2316 seems can not be closed yet cause the dashboard has not supported the filter feature.

**Pull request merged**